info.md.erb

```diff
@@ -4,20 +4,32 @@
   rescue
     data = {}
   end
-  version = data['version'].split('/').last
+
+  version = data['version'].to_s
   switch = data['jupyterlab_switch'].to_s
-  use_extension = data['jupyterlab_use_extension'].to_s
-  extension_enabled = false
-  if switch == "1" and use_extension == "1" and version.split('.').first.to_i > 0
-    extension_enabled = true
+  use_extension = data['use_extension'].to_s
+  classroom = data['classroom'].to_s
+
+  def jupyterlab_dir
+    base = "#{OodSupport::User.new.dir}/.juypter/lab"
+    match = JSON.parse(user_defined_context_file.read)['version'].to_s.match('app_jupyter/(?<majmin>\d+\.\d+)')
+    majmin = if !match.nil? && match.named_captures.key?('majmin')
```

Comment:
I thought this was a typo at first, `majmin`. Maybe `maj_min_version` or `two_digit_version` is better?

Also you should just be able to use `data['version'].to_s` on 15 instead of re-reading the file.

Reply:
> I thought this was a typo at first, `majmin`. Maybe `maj_min_version` or `two_digit_version` is better?

I will change it accordingly.


> Also you should just be able to use `data['version'].to_s` on 15 instead of re-reading the file.

I did it but it was not working. Here is the error message:
```
Error when rendering info view: NameError - undefined local variable or method `data' for #<BatchConnect::Session:0x00007fea1449ab28>
```

Reply:
Oh i see. Yea, then pass it in through `def jupyterlab_dir(version)`. Reading files are dangerous and slow, so we need to be sure it's safe. I'm guessing if this card breaks - it break the entire card so folks can't connect. (though as you have the error, you may be able to confirm what happens when this card errors).